content/blogs/newsletter-104.md

```diff
@@ -0,0 +1,68 @@
+
+The aim of this effort is to bring the improvements and fixes from WINE that help with the support of NT6+ software that otherwise wouldn't have worked.
+This also helps getting rid of hacks introduced.
+
+## Audio improvements
+Several major improvements in audio subsystem have been made by Oleg Dubinskiy recently.
+In particular, he implemented handling modern audio formats (e.g., Extensible audio, Float audio and others), [PR #6686](https://github.com/reactos/reactos/pull/6686), which support high sample rates, bits per sample sound quality and more than two (stereo) output channels.
+
+{{< video src="/img/blogs/newsletter-104/ros_extensible_format.mp4" type="video/mp4" caption="Testing the extensible audio format of some apps">}}
+
+This allows many 3rd-party programs and games (e.g., AIMP 5.30, QMMP 0.12.17, all Chromium based browsers and GameDevTycoon game) which are using modern formats, to open an audio device and play the sound properly. However, for now the sound is usually working when the audio miniport driver (e. g., Intel AC97, Realtek HD Audio etc.) supports the requested format directly.
+Otherwise, it won't work, because playing the format which is not supported by driver needs resampling support, which is currently not finished, so is disabled for now in ReactOS by default.
+
+Another improvement he made, is a support for looped wave playback ([PR #6761](https://github.com/reactos/reactos/pull/6761)).
+Looped playback functionality allows to play some audio data multiple times (in other words, play it in loop). The audio data to stream is contained in wave headers (each portion of it is in its own `WAVEHDR` structure).
+
+{{< video src="/img/blogs/newsletter-104/BRD_test.mp4" type="video/mp4" caption="Testing the looped wave playback in action">}}
+
+In Windows, this is supported by `WHDR_BEGINLOOP` and `WHDR_ENDLOOP` flags together with specifying the amount of loops to play the audio data in `WAVEHDR.dwLoops` member. As a single wave header, same multiple headers can be looped there. Oleg has implemented only looping a single wave header, while looping two or more wave headers is not supported yet. However we have no any testcase for the second case yet, because Oleg did not find any apps those request to loop multiple headers. But, as it was tested with BRD Demo app, looping a single header works properly as it should.
+
+Besides that, he is also working on some another audio improvements (not in the tree yet). For example, `GetWavePosition` API implementation, which retrieves the current wave playback position in time, and is required by many apps to play the sound correctly too. Another improvement is a some portion of fixes in DirectSound `IKsPropertySet` implementation, which allows to properly detect sound card by VirtualDJ 8.2, same as it's used by Microsoft DirectX Diagnostic Tool as well, and allows to display more information by audio device there. These and more fixes are coming soon!
```

Comment:
```suggestion
Besides that, he is also working on some another audio improvements (not in the tree yet).
For example, `GetWavePosition` API implementation, which retrieves the current wave playback position in time, and is required by many apps to play the sound correctly too.
Another improvement is a some portion of fixes in DirectSound `IKsPropertySet` implementation, which allows to properly detect sound card by VirtualDJ 8.2, same as it's used by Microsoft DirectX Diagnostic Tool as well, and allows to display more information by audio device there. These and more fixes are coming soon!
```

Should we announce things not in tree yet?
I don't think we usually do this.

Reply:
But why can't we? Maybe it's also a good idea, moreover if these thing will be added in tree soon.

Reply:
The input was coming from @oleg-dubinskiy and he said his report isn't finished yet. Perhaps changing `Besides that, he is also working on some another audio improvements (not in the tree yet).` to something like `His future plan is to further improve the audio stack with more fixes` or the like would probably make sense.

Reply:
Yeah, seems such a rephrasing sounds better in this case.